src/widget/wpushbutton.cpp

```diff
@@ -142,10 +150,16 @@ void WPushButton::setup(const QDomNode& node, const SkinContext& context) {
     }
 
     if (leftConnection) {
+        const ConfigKey& configKey = leftConnection->getKey();
+        if (configKey.item == "play" ||
+                configKey.item == "play_stutter" ||
+                configKey.item == "start_play") {
+            m_isPlayButton = true;
+            m_group = configKey.group;
+        }
```

Comment:
doing this is the class is really suboptimal. I would prefer to see an inheritance-based version that does not touch WPushButton this intrusively.

Reply:
I knew you'd say that : )

Reply:
I looked into this and the two ways to have a dedicated WPlayButton (I can think of) don't seem very elegant:
1) use `WPlayButton` in offical skins:
custom skins won't provide the drop feature
2) detect the 'play' left-click in the skin parser and instantiate a `WPlayButton`:
the WPushButton/WBaseWidget is instantiated _before_ its connections are parsed https://github.com/mixxxdj/mixxx/blob/d2e37ace7df609ed3518286123e3f96972683bbb/src/skin/legacy/legacyskinparser.cpp#L963
Even if possible, I don't think this feature justifies the (risky) refactoring the skin parser.

Do you have other ideas how to do it?

Reply:
I'd definitely prefer the first one. 3rd Party skin designers may not even like this feature, so we shouldn't implement it behind their backs. If they want to opt into it, the needed changes are likely trivial.

Reply:
Alright then, I'll move this to WPlayButton :+1: